test/unit/parser/format_parse_test.cpp

```diff
@@ -732,39 +732,43 @@ TEST_F(format_parse_test, issue1544)
 TEST_F(format_parse_test, is_option_set)
 {
     std::string option_value{};
+    std::vector<std::string> positional_options{};
 
-    auto parser = get_parser("-l", "hallo", "--foobar", "ballo", "--");
+    auto parser = get_parser("-l", "hallo", "--foobar", "ballo", "--", "--loo", "--bar");
+    // `--` signals the end of options!
+    parser.add_positional_option(positional_options, sharg::config{});
+    parser.add_option(option_value, sharg::config{.short_id = 'b', .long_id = "bar"});
     parser.add_option(option_value, sharg::config{.short_id = 'l', .long_id = "loo"});
     parser.add_option(option_value, sharg::config{.short_id = 'f', .long_id = "foobar"});
 
     // you cannot call option_is_set before parse()
     EXPECT_THROW(parser.is_option_set("foo"), sharg::design_error);
 
     EXPECT_NO_THROW(parser.parse());
+    EXPECT_EQ(positional_options.size(), 2u);
```

Comment:
If this is not given, the following 2 lines might be invalid accesses.
```suggestion
    ASSERT_EQ(positional_options.size(), 2u);
```

Reply:
Went for `EXPECT_EQ(positional_options, (std::vector<std::string>{"--loo", "--bar"}));`